presto-main/src/main/java/com/facebook/presto/operator/aggregation/BooleanMinMaxAggregationFunctions.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.NullableBooleanState;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.base.Throwables;
+
+import java.lang.invoke.MethodHandle;
+
+import static com.facebook.presto.spi.StandardErrorCode.INTERNAL_ERROR;
+
+public class BooleanMinMaxAggregationFunctions
+{
+    private BooleanMinMaxAggregationFunctions()
+    {
+    }
+
+    public static void input(final MethodHandle methodHandle, NullableBooleanState state, boolean value)
+    {
+        if (state.isNull()) {
+            state.setNull(false);
+            state.setBoolean(value);
+            return;
+        }
+        try {
+            if ((boolean) methodHandle.invokeExact(value, state.getBoolean())) {
+                state.setBoolean(value);
+            }
+        }
+        catch (Throwable t) {
+            Throwables.propagateIfInstanceOf(t, Error.class);
+            Throwables.propagateIfInstanceOf(t, PrestoException.class);
+            throw new PrestoException(INTERNAL_ERROR, t);
+        }
+    }
+
+    public static void combine(final MethodHandle methodHandle, NullableBooleanState state, NullableBooleanState otherState)
```

Comment:
Also see my comment above about re-using the input function instead of having this combine function